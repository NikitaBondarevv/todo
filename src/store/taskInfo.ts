import { createSlice } from '@reduxjs/toolkit'
import type { PayloadAction } from '@reduxjs/toolkit'

import { IInfo } from 'interfaces/IInfo'
import { TaskInfoState } from 'interfaces/ITaskInfoState'

const initialState: TaskInfoState = {
  data: {
    done: 0,
    inProgress: 0,
    total: 0,
    waiting: 0
  }
}

export const taskInfo = createSlice({
  name: 'taskInfo',
  initialState,
  reducers: {
    setInfo: (state: TaskInfoState, action: PayloadAction<IInfo>) => {
      state.data = action.payload
    }
  },
})

export const { setInfo } = taskInfo.actions

export default taskInfo.reducer
