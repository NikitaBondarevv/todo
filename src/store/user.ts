import { createSlice } from '@reduxjs/toolkit'
import type { PayloadAction } from '@reduxjs/toolkit'

import { IUser } from 'interfaces/IUser'

export interface UserState {
  data?: IUser
  isAuthenticated: boolean
}

const initialState: UserState = {
  isAuthenticated: false,
}

export const user = createSlice({
  name: 'user',
  initialState,
  reducers: {
    login: (state: UserState, action: PayloadAction<IUser>) => {
      state.data = action.payload
      state.isAuthenticated = true
    },
    
    logout: (state: UserState) => {
      delete state.data
      state.isAuthenticated = false
    },
  },
})

// Action creators are generated for each case reducer function
export const { login, logout } = user.actions

export default user.reducer
