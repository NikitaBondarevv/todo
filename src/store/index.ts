import { configureStore } from '@reduxjs/toolkit'
import { useDispatch, useSelector } from 'react-redux'
import type { TypedUseSelectorHook } from 'react-redux'

import user from './user'
import taskInfo from './taskInfo'

export const store = configureStore({
  reducer: {
    user: user,
    taskInfo: taskInfo
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch

export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
