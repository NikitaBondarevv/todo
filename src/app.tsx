import { useEffect } from 'react'

import { useAppDispatch } from './store'
import { checkUser } from 'contracts/checkUser'
import { Header } from 'components/header'
import { Pages } from 'pages'
import { login } from 'store/user'
import './styles.css'

export const App = () => {
  const dispatch = useAppDispatch()

  useEffect(() => {
    const fetchData = async () => {
      const user = await checkUser()

      if (!user.error) {
        dispatch(login(user))
      }
    }

    fetchData()
  }, [])

  return (
    <>
      <Header />
      <Pages />
    </>
  )
}
