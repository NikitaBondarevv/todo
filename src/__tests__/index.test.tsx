import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'

import { App } from '../app'
import { store } from 'store'

jest.mock('components/header', () => ({
  Header: () => <header data-testid="header" />
}))

jest.mock('pages', () => ({
  Pages: () => <main />
}))

jest.mock('contracts/checkUser', () => ({
  checkUser: () => ({})
}))


describe('<App />', () => {
  test('should render header and main', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </BrowserRouter>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
