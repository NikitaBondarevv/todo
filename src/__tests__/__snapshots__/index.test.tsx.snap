// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render header and main 1`] = `
<DocumentFragment>
  <header
    data-testid="header"
  />
  <main />
</DocumentFragment>
`;
